### PR TITLE
feat: add `merchant_name` field in the response body

### PR DESCRIPTION
**Pull request by Aprabhat19**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
Made some changes in the `core/payment_methods/cards.rs` and added a field of merchant_name to the response body of the List of Payment Methods as shown in the attachment below : 

Screenshot 2023-06-03 at 9 23 05 PM

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?
I tested it manually.

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Review by Narayanbhat166 — CHANGES_REQUESTED** [review]
Please follow the commit message conventions https://www.conventionalcommits.org/en/v1.0.0/

You can install [cocogito](https://github.com/cocogitto/cocogitto) and run the command
```shell
# Argument is the PR title
cog verify "feat:Added the merchant_name field in the response body" 
```

To know what needs to be changed in the PR title.

**Review by SanchithHegde — APPROVED** [review]

**Review by Narayanbhat166 — APPROVED** [review]

**Comment by Narayanbhat166** [comment]
Add a screenshot in the PR description as to what visible changes are in this PR

**Review by SanchithHegde — DISMISSED** [review]

File: crates/api_models/src/payment_methods.rs [hunk 1]
```diff
@@ -437,6 +437,7 @@ pub struct PaymentMethodListResponse {
         }
     ]
     ))]
+    pub merchant_name: Option<Encryptable<masking::Secret<String>>>,
```
Comment (Narayanbhat166) [inline crates/api_models/src/payment_methods.rs:440]: you can use this type alias 
https://github.com/juspay/hyperswitch/blob/6f1a1a3bcb7466738b018a773a5fe4adc5cf2887/crates/common_utils/src/crypto.rs#L444

File: crates/api_models/src/payment_methods.rs [hunk 2]
```diff
@@ -437,6 +437,8 @@ pub struct PaymentMethodListResponse {
         }
     ]
     ))]
+    pub merchant_name: OptionalEncryptableName,
```
Comment (SanchithHegde) [inline crates/api_models/src/payment_methods.rs:440]: The schema information and documentation comments annotated to this field are that of the `payment_methods` field. Could you please fix that, and add schema information and doc comments for the `merchant_name` field?

**Review by Narayanbhat166 — APPROVED** [review]

**Review by SanchithHegde — DISMISSED** [review]

**Review by Narayanbhat166 — DISMISSED** [review]

**Review by SanchithHegde — APPROVED** [review]

**Comment by SanchithHegde** [comment]
Going ahead and merging this, only the OpenAPI spec had to be updated in the recent conflict resolution.

**Pull request merged**